### PR TITLE
Update state recovery command

### LINKED ISSUE #7726: Update state recovery command
### Description

Update State recovery command based on [LIP0054](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md#state-recovery-command)
- Update [schema](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md#parameters) for params
- Update [verification](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md#verification)
- Update [execution](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md#execution)

### Acceptance Criteria

- Should have all the unit tests

### Additional Information

- Related LIP PR https://github.com/LiskHQ/lips/pull/172/
- Related LIP https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### What was the problem?

This PR resolves #7726 

### How was it solved?

- Updated state recovery on mainchain logic

### How was it tested?

- Updated unit tests

### CODE REVIEW

File: framework/src/modules/interoperability/mainchain/commands/state_recovery.ts [hunk 1]
```diff
@@ -54,7 +55,7 @@ export class StateRecoveryCommand extends BaseInteroperabilityCommand<MainchainI
 		if (!terminatedStateAccountExists) {
 			return {
 				status: VerifyStatus.FAIL,
-				error: new Error('The terminated account does not exist'),
+				error: new Error('The terminated account does not exist.'),
```
Comment (ishantiw) [inline framework/src/modules/interoperability/mainchain/commands/state_recovery.ts:58]:
```suggestion
				error: new Error('The terminated state account does not exist.'),
```

**Review by ishantiw — CHANGES_REQUESTED** [review]
I think some test cases are missing as follows,
- `Interoperability module cannot be recovered`
- `Module is not registered on the chain`
- `Module is not recoverable`
- `Recovered store value cannot be empty`

File: framework/src/modules/interoperability/mainchain/commands/state_recovery.ts [hunk 2]
```diff
@@ -63,18 +64,49 @@ export class StateRecoveryCommand extends BaseInteroperabilityCommand<MainchainI
...
+					error: new Error('Recovered store value cannot be empty.'),
+				};
+			}
 			queryKeys.push(entry.storeKey);
 			storeQueries.push({
-				key: entry.storeKey,
+				key: Buffer.from([...storePrefix, ...entry.substorePrefix, ...entry.storeKey]),
 				value: utils.hash(entry.storeValue),
 				bitmap: entry.bitmap,
 			});
```
[the remark this answers is not in the document]
Comment (mitsuaki-u) [reply framework/src/modules/interoperability/mainchain/commands/state_recovery.ts:112]: Updated error message but what does it have to do with the quoted line 109?

File: framework/src/modules/interoperability/types.ts [hunk 3]
```diff
@@ -385,3 +385,10 @@ export interface CCMRegistrationParams {
 	name: string;
 	messageFeeTokenID: Buffer;
 }
+
+export interface RecoverContext {
```
[the remark this answers is not in the document]
Comment (mitsuaki-u) [reply framework/src/modules/interoperability/types.ts:389]: It's not really return type as there is nothing returned from `recover` function

File: framework/src/modules/interoperability/mainchain/commands/state_recovery.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ishantiw) [reply framework/src/modules/interoperability/mainchain/commands/state_recovery.ts:112]: ah I couldn't add message at Line 126, thats why. but yeah just wanted to get error messages updated

File: framework/src/modules/interoperability/types.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ishantiw) [reply framework/src/modules/interoperability/types.ts:389]: then you have to update Line 143 where you typecasted it as `recover: () => RecoverContext;`, I think it should be `recover: (ctx: RecoverContext) => Promise<void>;`

**Review by Phanco — APPROVED** [review]

**Pull request merged**